sdk/tests/conformance/misc/expando-loss.html

```diff
@@ -0,0 +1,234 @@
+    var simpleData = [
+        {
+            creationFn: glProt.createTexture,
+            bindFn: glProt.bindTexture,
+            bindConstant: glProt.TEXTURE_2D,
+            retrieveConstant: glProt.TEXTURE_BINDING_2D,
+            name: "TEXTURE_BINDING_2D",
+        },
+        {
+            creationFn: glProt.createFramebuffer,
+            bindFn: glProt.bindFramebuffer,
+            bindConstant: glProt.FRAMEBUFFER,
+            retrieveConstant: glProt.FRAMEBUFFER_BINDING,
+            name: "FRAMEBUFFER_BINDING",
+        },
+        {
+            creationFn: glProt.createRenderbuffer,
+            bindFn: glProt.bindRenderbuffer,
+            bindConstant: glProt.RENDERBUFFER,
+            retrieveConstant: glProt.RENDERBUFFER_BINDING,
+            name: "RENDERBUFFER_BINDING",
+        },
+        {
+            creationFn: glProt.createBuffer,
+            bindFn: glProt.bindBuffer,
+            bindConstant: glProt.ELEMENT_ARRAY_BUFFER,
+            retrieveConstant: glProt.ELEMENT_ARRAY_BUFFER_BINDING,
+            name: "ELEMENT_ARRAY_BUFFER_BINDING",
+        },
+        {
+            creationFn: glProt.createBuffer,
+            bindFn: glProt.bindBuffer,
+            bindConstant: glProt.ARRAY_BUFFER,
+            retrieveConstant: glProt.ARRAY_BUFFER_BINDING,
+            name: "ARRAY_BUFFER_BINDING",
+        },
+        {
+            creationFn: glProt.createTexture,
+            bindFn: glProt.bindTexture,
+            bindConstant: glProt.TEXTURE_CUBE_MAP,
+            retrieveConstant: glProt.TEXTURE_BINDING_CUBE_MAP,
+            name: "TEXTURE_BINDING_CUBE_MAP",
+        },
+    ];
+
+    simpleData.forEach(function(test) {
+        var instance = test.creationFn.apply(gl, []);
+        var msg = "getParameter(" + test.name + ")";
+        setTestExpandos(instance);
+
+        test.bindFn.apply(gl, [test.bindConstant, instance]);
+        assertMsg(instance === gl.getParameter(test.retrieveConstant), msg + " returns instance that was bound.");
+
+        instance = null;
+        webglHarnessCollectGarbage();
+
+        verifyTestExpandos(gl.getParameter(test.retrieveConstant), msg);
+    });
+    debug('');
+}
+
+// Attach a couple of shaders to a program and verify no expando loss when you call
+// getAttachedShaders and getParameter(CURRENT_PROGRAM).
+function testProgramsAndShaders() {
+    debug('Programs and Shaders');
+
+    var vs = wtu.setupSimpleColorVertexShader(gl);
+    setTestExpandos(vs);
+
+    var fs = wtu.setupSimpleColorFragmentShader(gl);
+    setTestExpandos(fs);
+
+    var program = wtu.setupProgram(gl, [vs, fs]);
+    setTestExpandos(program);
+    assertMsg(program === gl.getParameter(gl.CURRENT_PROGRAM), "getParameter(gl.CURRENT_PROGRAM) return instance set with useProgram");
+
+    var attachedShaders = gl.getAttachedShaders(program);
+    assertMsg(vs === attachedShaders[0], "getAttachedShaders[0] returns instance set with attachShader");
```

Comment:
Agreed; this order doesn't seem to be specified in the ES 2.0 spec. Would be safer to write:

assertMsg(vs === attachedShaders[0] || vs === attachedShaders[1], ...);
assertMsg(fs === attachedShaders[0] || fs === attachedShaders[1], ...);